go/vumitools/app_worker.py

```diff
@@ -252,9 +253,19 @@ def find_outboundmessage_for_event(self, event):
 
     @inlineCallbacks
     def find_message_for_event(self, event):
-        outbound_message = yield self.find_outboundmessage_for_event(event)
-        if outbound_message:
-            returnValue(outbound_message.msg)
+        if "outbound_msg" in event.cache:
+            outbound_json = event.cache["outbound_json"]
+            if outbound_json is None:
+                returnValue(None)
+            returnValue(TransportUserMessage.from_json(outbound_json))
```

Comment:
I take it this specifically allows us to cache missing messages? If so, can you add some comments to that effect?